### PR TITLE
CHE-1039: add ability to bind additional folder to docker dev machine

### PR DESCRIPTION
**Pull request by garagatyi**
@skabashnyuk @eivantsov @vparfonov @evoevodin Please review

### CODE REVIEW

File: plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java [hunk 1]
```diff
@@ -121,11 +121,13 @@ public DockerInstanceProvider(DockerConnector docker,
         this.supportedRecipeTypes = Collections.singleton("dockerfile");
         this.projectFolderPath = projectFolderPath;
 
+        allMachinesSystemVolumes = filterEmptyAndNullValues(allMachinesSystemVolumes);
+        devMachineSystemVolumes = filterEmptyAndNullValues(devMachineSystemVolumes);
         if (SystemInfo.isWindows()) {
-            allMachinesSystemVolumes = escapePaths(allMachinesSystemVolumes);
-            devMachineSystemVolumes = escapePaths(devMachineSystemVolumes);
+            allMachinesSystemVolumes = escapePaths(filterEmptyAndNullValues(allMachinesSystemVolumes));
```
Comment (voievodin) [inline plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java]: Why did you add `filterEmptyAndNullValues` to the `escapePaths` method, as far as i understand they are already filtered 2 lines above + i think that it is better to name method `removeEmptyAndNullValues` which is more obvious against filter.

**Comment by skabashnyuk** [comment]
ok

File: plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java (hunk 1 above)
Comment (garagatyi) [inline plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java]: Oh, its a mistake. I forgot to remove that. Tnx

Comment (garagatyi) [inline plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java]: done

**Comment by voievodin** [comment]
ok
